### ISSUE TITLE
Enable Link-Time Optimization (LTO) for the Tauri part

### ISSUE THREAD

**Issue by zamazan4ik** [kmdreko/venator#2]
Hi!

I noticed that in the `Cargo.toml` [file](https://github.com/kmdreko/venator/blob/main/venator-app/src-tauri/Cargo.toml) Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

According to the official Tauri [documentation](https://v1.tauri.app/v1/guides/building/app-size/#rust-build-time-optimizations), you may also be interested in enabling other optimizations.

Thank you.

**Comment by kmdreko** [comment]
Thanks for bringing it up! Something I hadn't thought about.

You linked to the v1.0 docs but the [v2.0 docs on app size](https://v2.tauri.app/concept/size/) say the same thing. I'll skip `opt-level = "s"` (since I'm not interested in optimizing specifically for size) and `panic = "abort"` (for now, I may change to that in the future), but the rest look good:

```
[profile.release]
codegen-units = 1
lto = true
strip = true
```

**Comment by kmdreko** [comment]
I've published v0.2.1 which includes this change: [3e8ef79](https://github.com/kmdreko/venator/commit/3e8ef79ed98673553b32b2166156201074662a02)
